server/main.lua

```diff
@@ -20,4 +21,61 @@ for channel, jobs in pairs(restrictedChannels) do
         local player = exports.qbx_core:GetPlayer(source)
         return jobs[player.PlayerData.job.name] and player.PlayerData.job.onduty
     end)
-end
\ No newline at end of file
+end
+
+---@param source number
+local function deleteProp(source)
+    local prop = playerRadios[source]
+
+    if prop then
+        if DoesEntityExist(prop) then
+            DeleteEntity(prop)
+        end
+
+        playerRadios[source] = nil
+    end
+end
+
+---@param source number
+---@return number?
+lib.callback.register('qbx_radio:server:spawnProp', function(source)
+    local ped = GetPlayerPed(source)
+    local coords = GetEntityCoords(ped)
+    local object = CreateObject(`prop_cs_hand_radio`, coords.x, coords.y, coords.z, true, false, false)
+
+    local propExists = lib.waitFor(function()
+        if DoesEntityExist(object) then
+            return true
+        end
+    end, locale('failed_spawn'), 2000)
+
+    if not propExists then return end
+
+    playerRadios[source] = object
+
+    local netId = NetworkGetNetworkIdFromEntity(object)
+
+    SetEntityIgnoreRequestControlFilter(object, true)
+
+    return netId
+end)
+
+AddEventHandler('playerDropped', function()
+    local src = source
+
+    deleteProp(src)
+end)
+
+AddEventHandler('onResourceStop', function(resource)
+    if resource ~= cache.resource then return end
+
+    for i = 1, #playerRadios do
```

Comment:
```suggestion
    for i in ipairs(playerRadios) do
```
Player IDs won't necessarily start at 1 and some may be missing after the resource stops, since some players may have already left.